### PR TITLE
Add a DeployedCondition for the VirtualMachine object

### PR DESCRIPTION
**Pull request by sreyasn**
This change adds a conditionType for the deployment status of a virtualmachine on
the infrastructure provider.

### CODE REVIEW

**Review by janitha — APPROVED** [review]

File: api/v1alpha1/condition_consts.go [hunk 1]
```diff
@@ -37,6 +37,14 @@ const (
 	VirtualMachineImageNotFoundReason = "VirtualMachineImageNotFound"
 )
 
+const (
+	// DeployedCondition exposes the status of the VirtualMachine deployment.
```
Comment (aruneshpa) [inline api/v1alpha1/condition_consts.go:41]: I think the `VirtualMachine deployment` reads as `VirtualMachineDeployment`. Could you update this to reflect that we are talking about the VM being deployed on the infrastructure provider?

**Review by aruneshpa — APPROVED** [review]

File: api/v1alpha1/condition_consts.go (hunk 1 above)
Comment (sreyasn) [reply api/v1alpha1/condition_consts.go:41]: will do. Thanks for taking a look

File: api/v1alpha1/condition_consts.go [hunk 2]
```diff
@@ -37,6 +37,14 @@ const (
 	VirtualMachineImageNotFoundReason = "VirtualMachineImageNotFound"
 )
 
+const (
+	// DeployedCondition exposes the deployment status of a VirtualMachine on the infrastructure provider.
+	DeployedCondition ConditionType = "Deployed"
```
Comment (akutz) [inline api/v1alpha1/condition_consts.go:42]: I'm not sure I understand what this means. The VM hardware deployed successfully? The virtual hardware failed? The guest customization succeeded / failed? Both? I am not sure this makes sense as-is. Let's revisit this a bit more. I don't know that aggregate conditions make sense, and we already have a guest customization condition, so is deploy separate or a superset?

File: api/v1alpha1/condition_consts.go [hunk 3]
```diff
@@ -37,6 +37,14 @@ const (
...
+
+	// DeployFailedReason (Severity=Error) documents that the deployment of the VirtualMachine was not successful.
+	DeployFailedReason = "DeployFailed"
```
Comment (akutz) [inline api/v1alpha1/condition_consts.go:45]: Same as above.

**Review by akutz — CHANGES_REQUESTED** [review]

File: api/v1alpha1/condition_consts.go (hunk 2 above)
Comment (sreyasn) [reply api/v1alpha1/condition_consts.go:42]: Deploy is separate, It's synonymous the Created/Create in VM operator. 

guest customization is triggered in our Update path after Create/Deploy passes.

Comment (akutz) [reply api/v1alpha1/condition_consts.go:42]: > Deploy is separate, It's synonymous to Created/Create in VM operator.

The issue is that "deploy" is vague enough to be taken multiple ways. Plus, [conditions exist](https://maelvls.dev/kubernetes-conditions/) as a way to be more distinct than allowed by a top-level status field such as `phase`. So let's come up with a better word than `deployed` as it seems to generic and could be mistaken to mean guest customization as well.

**Pull request closed without merging**